src/Messages/OWSMessageSender.m

```diff
@@ -628,11 +629,17 @@ - (void)sendSyncTranscriptForMessage:(TSOutgoingMessage *)message
             // DEPRECATED - Remove after all clients have been upgraded.
             BOOL isLegacyMessage = ![message isKindOfClass:[OWSOutgoingSyncMessage class]];
 
-            NSDictionary *messageDict = [self encryptedMessageWithPlaintext:plainText
-                                                                toRecipient:recipient.uniqueId
-                                                                   deviceId:deviceNumber
-                                                              keyingStorage:[TSStorageManager sharedManager]
-                                                                     legacy:isLegacyMessage];
+            __block NSDictionary *messageDict;
+            // Mutating session state is not thread safe, so we operate on a serial queue, shared with decryption
+            // operations.
+            dispatch_sync([OWSDispatch sessionCipher], ^{
```

Comment:
My fear with documentation files is that the farther away they are from the code, the less likely they are to be kept up to date or read.

How do would you feel about my existing documentation here in the OWSDispatch class? https://github.com/WhisperSystems/SignalServiceKit/pull/98/files#diff-01540420599ea2ef7127ff1d06e00c81R17